source/agora/consensus/protocol/Nominator.d

```diff
@@ -1062,6 +1053,45 @@ extern(D):
         log.trace("valueExternalized: added slot id {} to ledger at height {}", height, last_height);
         () @trusted { this.fully_validated_value.clear(); }();
         () @trusted { this.seen_envs.clear(); }();
+        this.timers[TimersIdx.Externalize].rearm(ExternalizeTaskDelay, false);
+    }
+
+    /// If we have majority signatures then externalize to the ledger otherwise try again later
+    extern(D) private void checkExternalize () nothrow
+    {
+        try
+        {
+            if (pending_block.header.height == this.ledger.height + 1)
+            {
+                this.updateMultiSignature(pending_block.header);
+                if (!this.ledger.hasMajoritySignature(pending_block.header))
+                {
+
+                    log.dbg("{}: Will not yet externalize block #{} as we only have {} / {} signatures",
+                        __FUNCTION__, pending_block.header.height, pending_block.header.validators.setCount,
+                        pending_block.header.validators.count);
+                    this.timers[TimersIdx.Catchup].rearm(CatchupTaskDelay, false);
+                    this.timers[TimersIdx.Externalize].rearm(ExternalizeTaskDelay, false);
+                    return;
+                }
+                else
```

Comment:
Isn't there any case of `pending_block.header.height > this.ledger.height + 1` like in the middle of catch-up?

Reply:
No as we check in `valueExternalized` the current ledger height before creating a `pending_block`.